src/test/java/org/broadinstitute/hellbender/tools/reference/ReferenceSequenceTableUnitTest.java

```diff
@@ -0,0 +1,228 @@
+        return row1.getMd5().equals(row2.getMd5()) && row1.getLength() == row2.getLength();
+    }
+
+    @Test(dataProvider = "testQueryByMD5Data")
+    public void testQueryByMD5(String md5, int length, ReferenceSequenceTable table){
+        ReferenceSequenceTable.TableRow row = table.queryByMD5(md5);
+
+        Assert.assertEquals(row.getMd5(), md5);
+        Assert.assertEquals(row.getLength(), length);
+
+    }
+
+    @DataProvider(name = "testQueryBySequenceNameData")
+    public Object[][] testQueryBySequenceNameData() {
+        return new Object[][]{
+                // sequence name, expected number of rows, table
+                new Object[]{ "2", 2,
+                        tableGenerator(Arrays.asList(new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini.fasta"),
+                        new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini_chr2snp.fasta")),
+                        CompareReferences.MD5CalculationMode.USE_DICT),
+                },
+                new Object[]{ "1", 1,
+                        tableGenerator(Arrays.asList(new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini.fasta"),
+                                        new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini_chr2snp.fasta")),
+                                CompareReferences.MD5CalculationMode.USE_DICT),
+                },
+                new Object[]{ "5", 0,
+                        tableGenerator(Arrays.asList(new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini.fasta"),
+                                        new GATKPath("src/test/resources/org/broadinstitute/hellbender/tools/reference/CompareReferences/hg19mini_chr2snp.fasta")),
+                                CompareReferences.MD5CalculationMode.USE_DICT),
+                },
+        };
+    }
+
+
+    @Test(dataProvider = "testQueryBySequenceNameData")
+    public void testQueryBySequenceName(String expectedSequenceName, int expectedRows, ReferenceSequenceTable table){
+        Set<ReferenceSequenceTable.TableRow> rows = table.queryBySequenceName(expectedSequenceName);
+
+        if(expectedRows == 0){
+            Assert.assertTrue(rows.isEmpty());
+        } else {
+            Assert.assertEquals(rows.size(), expectedRows);
+
+            boolean contigPresent = false;
+            for (ReferenceSequenceTable.TableRow row : rows) {
+                contigPresent = false;
+                for (ReferenceSequenceTable.TableEntry entry : row.getEntries()) {
+                    if (entry.getColumnValue().equals(expectedSequenceName)) {
+                        contigPresent = true;
+                        continue;
+                    }
+                }
+                Assert.assertTrue(contigPresent);
```

Comment:
Add a String as a second arg to `assertTrue()` that explains what went wrong when the assert fails. Eg., 
```
"Contig " + expectedSequenceName + " not found in all rows returned by queryBySequenceName(expectedSequenceName)"
```
This will get output when the assert fails